src/managers/tls.py

```diff
@@ -52,37 +52,74 @@ def set_certificate(self) -> None:
 
     def set_truststore(self) -> None:
         """Creates the unit Java Truststore and adds the unit CA."""
-        keytool_cmd = "charmed-zookeeper.keytool" if self.substrate == "vm" else "keytool"
-
         try:
-            self.workload.exec(
-                command=[
-                    keytool_cmd,
-                    "-import",
-                    "-v",
-                    "-alias",
-                    "ca",
-                    "-file",
-                    self.workload.paths.ca,
-                    "-keystore",
-                    self.workload.paths.truststore,
-                    "-storepass",
-                    self.state.unit_server.truststore_password,
-                    "-noprompt",
-                ],
-            )
+            self._import_ca_in_truststore()
+
             if self.substrate == "vm":
                 self.workload.exec(
                     command=["chown", "snap_daemon:root", self.workload.paths.truststore],
                 )
-
         except (subprocess.CalledProcessError, ops.pebble.ExecError) as e:
             if "already exists" in str(e.stdout):
+                # replace ca in four steps to prevent the truststore for being empty at any point.
+                try:
+                    if self.substrate == "vm":
+                        self.workload.exec(
+                            command=["chown", "root:root", self.workload.paths.truststore],
+                        )
+                    self._import_ca_in_truststore("ca-temp")
+                    self._delete_ca_in_truststore("ca")
+                    self._import_ca_in_truststore("ca")
+                    self._delete_ca_in_truststore("ca-temp")
```

Comment:
**nitpick:** Maybe `ca-backup` or `ca-old`, something like that

Reply:
**suggestion:** Consider leading these lines with a comment outlining why we do this. It's 'obvious', sorta, but I think it would be a good comment for the long-term.

Reply:
Yes, absolutely will do

Reply:
Added some context in 67edf12b2844. I used 'ca-upcoming' instead of 'ca-temp' for clarity, but not any of your suggestions, since it's the new one that we import